### PR TITLE
write a more complete README

### PR DESCRIPTION
**Pull request by doudou**

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -14,11 +16,127 @@ device drivers, even for protcols that are naturally packet-based such as
...
+The return value is fully documented within the method documentation. Check it
+out for a complete specification.
+
+~~~cpp
+int extractPacket(uint8_t const* buffer, int buffer_size) {
+    if (not enough bytes in buffer to contain a start code) {
+        return 0; // wait for new bytes
+    }
+
+    if (!buffer starts with packet start code) {
```
Comment (g-arjones) [inline README.md]: For clarity: `if (buffer does not start with a packet start code) {`

File: README.md [hunk 2]
```diff
@@ -14,11 +16,127 @@ device drivers, even for protcols that are naturally packet-based such as
...
+~~~
+
+## Supported URIs
+
+- serial ports are open with `serial:/${DEVICE_PATH}[:BAUDRATE]`, e.g.
+  `serial:///dev/ttyUSB0:115200`
+- a UDP server is open with `udpserver://[PORT]`. The UDP server will
+  receive from any source, and send to the IP/Port of the last received
+  packet it received
+- a UDP client is open with `udp://host:PORT`. The UDP client will
```
Comment (g-arjones) [inline README.md]: I would document this one as `udp://host:outport[:inport]` instead

File: README.md (hunk 1 above)
Comment (doudou) [reply README.md]: done

File: README.md (hunk 2 above)
Comment (doudou) [reply README.md]: used `:remote_port[:local_port]`, I feel it easier to understand

**Pull request merged**